### PR TITLE
Silence warnings for NDPermuter file

### PR DESCRIPTION
**Pull request by mlxd**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:** This silences the warnings issued from codefactor for the NDPermuter class.

**Description of the Change:** As above.

**Benefits:** Prevents false warnings in other PRs

**Possible Drawbacks:** May miss opportunity for improvements to this class.

**Related GitHub Issues:**

### CODE REVIEW

File: pennylane_lightning/core/src/utils/NDPermuter.hpp [hunk 1]
```diff
@@ -294,4 +295,5 @@ template <size_t BLOCKSIZE = 1024> class DefaultPermuter {
 };
 
 } // namespace Pennylane::Util
-// LCOV_EXCL_END
+  // LCOV_EXCL_END
+  // NOLINTEND
```
Comment (maliasadi) [inline pennylane_lightning/core/src/utils/NDPermuter.hpp:299]:
```suggestion
  // NOLINTEND

```

**Review by maliasadi — APPROVED** [review]

File: pennylane_lightning/core/src/utils/NDPermuter.hpp (hunk 1 above)
Comment (multiphaseCFD) [reply pennylane_lightning/core/src/utils/NDPermuter.hpp:299]: Nice! Was also about to mark this.

**Review by multiphaseCFD — APPROVED** [review]
Thanks @mlxd !

File: pennylane_lightning/core/src/utils/NDPermuter.hpp (hunk 1 above)
Comment (maliasadi) [inline pennylane_lightning/core/src/utils/NDPermuter.hpp:299]:
```suggestion
// LCOV_EXCL_END
// NOLINTEND
```

**Pull request merged**